src/main/java/org/czekalski/petapiintegration/resourceServer/controller/AnimalController.java

```diff
@@ -0,0 +1,25 @@
+package org.czekalski.petapiintegration.resourceServer.controller;
+
+import org.czekalski.petapiintegration.resourceServer.apiResource.v1.dto.AnimalResourcesListDto;
+import org.czekalski.petapiintegration.resourceServer.service.AnimalService;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AnimalController {
+    private final AnimalService animalService;
+
+    public AnimalController(AnimalService animalService) {
+        this.animalService = animalService;
+    }
+
+    @GetMapping("/dogs/{cityId}/{stateId}")
+    public AnimalResourcesListDto getDogsFromCity(@PathVariable String cityId, @PathVariable String stateId, @RequestParam int size) {
```

Comment:
Chciałbym, żebyś wykorzystał `Page`, a nie customowy typ. Jak jakieś pola się nie zgadzają ze specyfikacją to mój błąd. Miało być tak, żeby `Page` pasował

Reply:
Tzn. miałem problem z tym aby wykorzystać Page bo to jest sprzężone z bazą danych z której Spring korzysta. Page należy do modułu Spring Data.